### PR TITLE
Allow additional markdown options

### PR DESCRIPTION
**Pull request by NickColley**
I think turning on new features may be too controversial so I'm adding functionality so the user can set their own preferences.

See https://github.com/alphagov/tech-docs-gem/issues/169 for why I want to do this.

### CODE REVIEW

**Comment by nickcolley** [comment]
Can we do this via the config.yaml file?

File: lib/govuk_tech_docs.rb [hunk 1]
```diff
@@ -40,15 +43,19 @@ def self.configure(context, options = {})
...
+      renderer: TechDocsHTMLRenderer.new(
+        with_toc_data: true,
+        api: true,
+        context: context,
+      ),
+      fenced_code_blocks: true,
+      tables: true,
+      no_intra_emphasis: true
+    }
+    markdown_options = context.config[:tech_docs][:markdown].transform_keys(&:to_sym) || {}
```
Comment (36degrees) [inline lib/govuk_tech_docs.rb]: What happens if there are no markdown options specified in the config file? Does this need to be guarded?

Comment (nickcolley) [reply lib/govuk_tech_docs.rb]: I believe It'll return falsy and the {} will be used.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 ## Unreleased
 
+### Fixes
```
Comment (nickcolley) [inline CHANGELOG.md]: this was accidentally added...

File: lib/govuk_tech_docs.rb (hunk 1 above)
Comment (36degrees) [reply lib/govuk_tech_docs.rb]: Doesn't seem to like the chaining when I try it:

```
irb(main):004:0> foo = { :tech_docs => { :something => true }}
=> {:tech_docs=>{:something=>true}}
irb(main):005:0> foo[:tech_docs][:markdown].transform_keys(&:to_sym) || {}
Traceback (most recent call last):
        4: from /Users/oliverbyford/.rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        3: from /Users/oliverbyford/.rbenv/versions/2.6.3/bin/irb:23:in `load'
        2: from /Users/oliverbyford/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        1: from (irb):5
NoMethodError (undefined method `transform_keys' for nil:NilClass)
```

Looks like it works OK with safe navigation (`&.`) though:

```
irb(main):004:0> foo = { :tech_docs => { :something => true }}
=> {:tech_docs=>{:something=>true}}
irb(main):006:0> foo[:tech_docs][:markdown]&.transform_keys(&:to_sym) || {}
=> {}
irb(main):007:0> foo = { :tech_docs => { :something => true, :markdown => { :be_awesome => true }}}
=> {:tech_docs=>{:something=>true, :markdown=>{:be_awesome=>true}}}
irb(main):008:0> foo[:tech_docs][:markdown]&.transform_keys(&:to_sym) || {}
=> {:be_awesome=>true}
```

**Comment by nickcolley** [comment]
Gonna pause this, write some cards and we can revisit if it makes sense.

**Pull request closed without merging**